src/Grpc.AspNetCore.Server/Internal/HttpContextServerCallContext.cs

```diff
@@ -74,7 +74,15 @@ protected override Metadata RequestHeadersCore
 
                     foreach (var header in HttpContext.Request.Headers)
                     {
-                        if (!header.Key.StartsWith(':'))
+                        if (header.Key.StartsWith(':'))
+                        {
+                            continue;
+                        }
+                        else if (header.Key.EndsWith(Metadata.BinaryHeaderSuffix))
```

Comment:
Ordinal compare. Is it case sensitive? Add tests either way.

Reply:
Good point, it's case sensitive in the Grpc.Core implementation. Will add a test.

Reply:
I recall someone saying that Metadata are always lower case. Is that right?

Reply:
Yup, https://github.com/grpc/grpc-dotnet/pull/69#discussion_r254132086, I misremembered and was thinking of something else.